test/test.make

```diff
@@ -0,0 +1,117 @@
+# Intel while reviewing the components.
+.PHONY: test_runtime_deps
+test: test_runtime_deps
+
+test_runtime_deps:
+	@ if ! diff -c \
+		runtime-deps.csv \
+		<( $(RUNTIME_DEPS) ); then \
+		echo; \
+		echo "runtime-deps.csv not up-to-date. Update RUNTIME_DEPS in test/test.make, rerun, review and finally apply the patch above."; \
+		false; \
+	fi
+
+RUNTIME_DEPS =
+
+# We use "go list" because it is readily available. A good replacement
+# would be godeps. We list dependencies recursively, not just the
+# direct dependencies.
+RUNTIME_DEPS += go list -f '{{ join .Deps "\n" }}' ./cmd/pmem-csi-driver |
+
+# This focuses on packages that are not in Golang core.
+RUNTIME_DEPS += grep '^github.com/intel/pmem-csi/vendor/' |
+
+# Filter out some packages that aren't really code.
+RUNTIME_DEPS += grep -v -e 'github.com/container-storage-interface/spec' |
+RUNTIME_DEPS += grep -v -e 'google.golang.org/genproto/googleapis/rpc/status' |
+
+# Reduce the package import paths to project names + download URL.
+# - strip prefix
+RUNTIME_DEPS += sed -e 's;github.com/intel/pmem-csi/vendor/;;' |
+# - use path inside github.com as project name
+RUNTIME_DEPS += sed -e 's;^github.com/\([^/]*\)/\([^/]*\).*;github.com/\1/\2;' |
+# - everything from gRPC is one project
+RUNTIME_DEPS += sed -e 's;google.golang.org/grpc/*.*;grpc-go,https://github.com/grpc/grpc-go;' |
+# - various other projects
+RUNTIME_DEPS += sed \
+	-e 's;github.com/google/uuid;google uuid,https://github.com/google/uuid;' \
+	-e 's;github.com/golang/protobuf;golang-protobuf,https://github.com/golang/protobuf;' \
+	-e 's;github.com/gogo/protobuf;gogo protobuf,https://github.com/gogo/protobuf;' \
+	-e 's;github.com/golang/glog;glog,https://github.com/golang/glog;' \
+	-e 's;github.com/pkg/errors;pkg/errors,https://github.com/pkg/errors;' \
+	-e 's;github.com/vgough/grpc-proxy;grpc-proxy,https://github.com/vgough/grpc-proxy;' \
+	-e 's;golang.org/x/.*;Go,https://github.com/golang/go;' \
+	-e 's;k8s.io/.*;kubernetes,https://github.com/kubernetes/kubernetes;' \
+	-e 's;gopkg.in/fsnotify.*;golang-github-fsnotify-fsnotify,https://github.com/fsnotify/fsnotify;' \
+	| cat |
+
+# Ignore duplicates.
+RUNTIME_DEPS += sort -u
+
+# E2E testing relies on a running QEMU test cluster. It therefore starts it,
+# but because it might have been running already and might have to be kept
+# running to debug test failures, it doesn't stop it
+.PHONY: test_e2e
+test_e2e: start
+	KUBECONFIG=`pwd`/_work/clear-kvm-kube.config REPO_ROOT=`pwd` go test -timeout 0 ./test/e2e
+
+.PHONY: run_tests
+test: run_tests
+run_tests: pmem-csi-driver _work/pmem-ca/.ca-stamp _work/evil-ca/.ca-stamp
+	TEST_WORK=$(abspath _work) \
+	$(TEST_CMD) $(shell go list $(TEST_ARGS) | sed -e 's;$(IMPORT_PATH);.;')
+
+_work/%/.ca-stamp: test/setup-ca.sh _work/.setupcfssl-stamp
+	rm -rf $(@D)
+	WORKDIR='$(@D)' PATH='$(PWD)/_work/bin/:$(PATH)' CA='$*' EXTRA_CNS="wrong-node-controller" $<
+	touch $@
+
+ _work/.setupcfssl-stamp:
```

Comment:
minor nit: there's an extra space before `_work/.setupcfssl-stamp:`.

Reply:
I fixed it.